python/cudf/cudf/core/column/lists.py

```diff
@@ -261,7 +262,7 @@ def as_string_column(
         # Call libcudf to format the list column
         return format_list_column(lc, separators)
 
-    def _transform_leaves(self, func, *args, **kwargs):
+    def _transform_leaves(self, func, *args, **kwargs) -> Self:
```

Comment:
IIUC this change was what motivated this update. What is the difference between using this and a bound generic like `T = TypeVar("T", bound="DataFrame")` as we've usually done?

Reply:
[PEP 673](https://peps.python.org/pep-0673/) has all the gory details. tl;dr this is new syntax for the old `T = TypeVar("T", bound="Foo")` setup, but more natural (one doesn't also have to annotate `self : T`, and it also works with classmethods where previously you would have written `def(cls: type[T], ...) -> T`.

Reply:
Agreed, no need to do it all in this PR.